core/types/block.libevm.go

```diff
@@ -146,6 +146,10 @@ func (b *Body) EncodeRLP(dst io.Writer) error {
 	})
 }
 
+func (b *Body) hasOptionalFields() bool {
```

Comment:
```suggestion
func (b *Body) hasOptionalFieldsSet() bool {
```

Reply:
Renamed to `hasAnyOptionalFieldSet` since it should be true if any (not all) optional field is set.